fgpyo/util/metric.py

```diff
@@ -322,3 +332,29 @@ def fast_concat(*inputs: Path, output: Path) -> None:
             io.write_lines(
                 path=output, lines_to_write=list(io.read_lines(input_path))[1:], append=True
             )
+
+    @staticmethod
+    def read_header(
+        reader: io.Reader,
+        comment_char: str = DEFAULT_HEADER_COMMENT_CHAR,
+    ) -> List[str]:
+        """
+        Read the header from an open file.
+
+        Comment and empty lines will be ignored.
+
+        Args:
+            reader: An open, readable file
+            comment_char: The character which indicates the start of a comment line.
+
+        Returns:
+            A list of field names found in the header line.
+        """
+
+        for line in reader:
+            if not line.startswith(comment_char) and not line.strip() == "":
+                break
+        else:
+            raise ValueError("No header found")
+
+        return line.rstrip("\r\n").split("\t")
```

Comment:
I noticed `Metric` always assumes a tab-delimited file.

Would it be sensible to provide the delimiter as an argument to the `read()` and `write()` methods?

Reply:
I wouldn't object to it, so long as the default is always `tab`.  However, that does raise the question of whether we should then do better at quoting/escaping since tabs don't often show up in values, but commas and spaces will.

Reply:
Good thought - I'll open a separate issue to address custom delimiters